Wrappers/Python/cil/optimisation/algorithms/PDHG.py

```diff
@@ -166,11 +293,107 @@ def update(self):
 
         self.g.proximal(self.x_tmp, self.tau, out=self.x)
 
-        #update_previous_solution() called after update by base class
-        #i.e current solution is now in x_old, previous solution is now in x
+        # update_previous_solution() #called after update by base class
+        #i.e current solution is now in x_old, previous solution is now in x        
+
+        # update the step sizes for special cases
+        self.update_step_sizes()
+
+
+    def set_step_sizes(self, sigma=None, tau=None):
+
+        """Default step sizes for the PDHG algorithm"""
+        
+        # Compute operator norm
+        self.norm_op = self.operator.norm()
+
+        if tau is None and sigma is None:            
+            self._sigma = 1./self.norm_op
+            self._tau = 1./self.norm_op
+        elif tau is None:
+            self._tau = 1./(self.sigma*self.norm_op**2)
+        elif sigma is None:
+            self._sigma = 1./(self.tau*self.norm_op**2)
+        else:
+            self._sigma = sigma
+            self._tau = tau
+      
+    def update_step_sizes(self):
+
+        r"""
+        
+        Updates the step sizes :math:`\sigma` and :math:`\tau` and :math:`\theta` in the cases of primal or dual acceleration using the strongly convexity property.
+        
+        - :math:`g` is strongly convex with constant :math:`\gamma=`  ``gamma_g``.
+        - :math:`f^{*}` is strongly convex with constant :math:`\gamma=`  ``gamma_fconj``.
+
+        Note
+        ----
+
+        The case where both functions are strongly convex is not available at the moment.  
+        
+
+        .. todo:: Implement acceleration of PDHG when both functions are strongly convex.
+                 
+        """
+
+        # Update sigma and tau based on the strong convexity of G
+        if self.gamma_g is not None:
+            self.theta = 1.0/ np.sqrt(1 + 2 * self.gamma_g * self.tau)
+            self._tau *= self.theta
+            self._sigma /= self.theta 
+
+        # Update sigma and tau based on the strong convexity of F
+        # Following operations are reversed due to symmetry, sigma --> tau, tau -->sigma
+        if self.gamma_fconj is not None:            
+            self.theta = 1.0 / np.sqrt(1 + 2 * self.gamma_fconj * self.sigma)
+            self._sigma *= self.theta
+            self._tau /= self.theta    
```

Comment:
This is a question: if one passes both `gamma_g` and `gamme_fconj` the new `sigma` and `tau` will be updated two times by a `theta` which is (likely to be) different. 

Let me rewrite it like

```python
if self.gamma_g is not None:
    theta_g = 1.0/ np.sqrt(1 + 2 * self.gamma_g * self.tau)
    self._tau *= theta_g
    self._sigma /= theta_g

# Update sigma and tau based on the strong convexity of F
# Following operations are reversed due to symmetry, sigma --> tau, tau -->sigma
if self.gamma_fconj is not None:            
    theta_fconj = 1.0 / np.sqrt(1 + 2 * self.gamma_fconj * self.sigma)
    self._sigma *= theta_fconj 
    self._tau /= theta_fconj 

if self.gamma_fconj is not None:
    self.theta = theta_fconj
else:
    self.theta = theta_g
```


It looks like then that the `gamma_fconj` definition of `theta` is prevails wrt to the one coming from `g`. 

Questions:

1. is this the intended behaviour?
2. is `self.theta` correct as it is now?